target-app/Home/Views/HomeViewController.swift

```diff
@@ -24,22 +45,38 @@ class HomeViewController: UIViewController {
         setupMapConstraints()
     }
     
+    override func viewDidAppear(_ animated: Bool) {
+        LocationManager.shared.requestLocationAuthorization()
+        
+        if CLLocationManager.locationServicesEnabled() {
+            locationManager.delegate = self
+            locationManager.desiredAccuracy = kCLLocationAccuracyBest
+            locationManager.startUpdatingLocation()
+        }
+        
+        mapView.delegate = self
+        
+        if let coor = mapView.userLocation.location?.coordinate{
+            mapView.setCenter(coor, animated: true)
+        }
+    }
+    
     func setupNavigationBar() {
         navigationController?.setNavigationBarHidden(false, animated: true)
         
         navigationItem.title = "home_title".localized
         
         let barLeftButtonItem = UIBarButtonItem(image: UIImage(named: "ic_home_profile"),
-                                            style: .plain,
-                                            target: self,
-                                            action: nil)
+                                                style: .plain,
+                                                target: self,
+                                                action: nil)
```

Comment:
Just and my OCD but this could look prettier like:


```swift
let barLeftButtonItem = UIBarButtonItem(
  image: UIImage(named: "ic_home_profile"),
  style: .plain,
  target: self,
  action: nil
)
```

Reply:
done!